src/cdomains/listDomain.ml

```diff
@@ -2,6 +2,8 @@ open Cil
 
 module GU = Goblintutil
 
```

Comment:
For domains as generic as lists, I don't think it's an issue if there are unused functions. These form the future building blocks of other domains, so I'd argue we keep these.

Reply:
Sure, but it is suspicious if operations are defined but never used, even for such a simple list domain. We have that in our `ValueDomain` as a variant, but I think I've never seen that list abstraction in our analysis results. It just looks like incomplete beginnings of some analysis that never panned out. Maybe @vesalvojdani has an idea?

Because the more involved shape analysis was removed in #474, so I have no idea what, if anything, this still does. I'm fine with keeping these for now, but it'd be good to have the comment here pointing out that it's not fully used, such that anyone looking at this code doesn't get the wrong idea that this is something significant and maintained.

Reply:
Turns out we only use this when `exp.list-type` is enabled. There's only one test for that: it's skipped and if executed fails for 5/6 annotations, which deepens my doubts about it.

Reply:
Completely removed after GobCon decision.